src/couch/src/couch_httpd_auth.erl

```diff
@@ -253,6 +246,68 @@ jwt_authentication_handler(Req) ->
             Req
     end.
 
+get_roles_claim(Claims) ->
+    Roles_Claim_Name_Param = config:get(
+        "jwt_auth", "roles_claim_name"
+    ),
+    Roles_Claim_Path_Param = config:get(
+        "jwt_auth", "roles_claim_path"
+    ),
+    Roles =
+        case Roles_Claim_Name_Param of
+            undefined ->
+                % Check for implicitly role_claim "_couchdb.roles" in JWT token
+                case lists:keyfind(<<"_couchdb.roles">>, 1, Claims) of
+                    false ->
+                        % `_couchdb.roles` not found in token
+                        % check if `roles_claim_path` is set
+                        case Roles_Claim_Path_Param of
+                            undefined ->
+                                [];
+                            _ ->
+                                % no implicit or explicit value for `_couchdb.roles` found, use `roles_claim_path`
+                                Roles_Claim_Path = [
+                                    ?l2b(Item)
+                                 || Item <- string:tokens(Roles_Claim_Path_Param, ".")
+                                ],
+                                couch_util:get_nested_json_value({Claims}, Roles_Claim_Path)
+                        end;
+                    {_, Token_CouchDBRoles} ->
+                        % `_couchdb.roles` found in token, read value
+                        couch_log:info(
+                            "Implicit value for `_couchdb.roles` found and used in token, please migrate to" ++
+                                " `roles_claim_path`!",
+                            []
+                        ),
+                        Token_CouchDBRoles
+                end;
+            _ ->
+                % other value for `roles_claim_name` found.
+                if
+                    Roles_Claim_Path_Param =/= undefined ->
+                        couch_log:info(
```

Comment:
logging this on every occurrence is not great. You've also decided that roles_claim_name is deprecated but there's been no suggestion of that so far and it doesn't seem reasonable to me.

Reply:
It was only a [suggestion](https://github.com/apache/couchdb/pull/4041#discussion_r884104705) from me, because I think it's better to only have one param instead of two. We only have to find a find a common denominator...